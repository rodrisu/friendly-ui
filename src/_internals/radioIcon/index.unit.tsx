import React from 'react'
import renderer from 'react-test-renderer'

import { RadioIcon } from './index'

describe('RadioIcon', () => {
  it('Should render an empty circle by default', () => {
    const radio = renderer.create(<RadioIcon />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a check icon if isChecked is set to true', () => {
    const radio = renderer.create(<RadioIcon isChecked />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a loader if isLoading is set to true', () => {
    const radio = renderer.create(<RadioIcon isLoading />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a loader if isLoading and isChecked are set to true', () => {
    const radio = renderer.create(<RadioIcon isChecked isLoading />).toJSON()
    expect(radio).toMatchSnapshot()
  })
})
