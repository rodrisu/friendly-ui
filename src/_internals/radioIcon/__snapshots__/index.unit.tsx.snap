// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioIcon Should render a check icon if isChecked is set to true 1`] = `
.c0 {
  fill: #00AFF5;
}

.c0.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c0.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c0 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c1 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c1.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c1 circle.outer {
  stroke: #00AFF5;
}

.c1 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c1.thin circle {
  stroke-width: 3;
}

.c1.absolute {
  position: absolute;
}

<svg
  aria-hidden={true}
  className="kirk-icon kirk-icon-circle c0 c1"
  height={24}
  viewBox="0 0 66 66"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  
  <circle
    className="outer"
    cx="33"
    cy="33"
    fill="none"
    r="30"
  />
  <circle
    className="inner"
    cx="33"
    cy="33"
    r="18"
  />
</svg>
`;

exports[`RadioIcon Should render a loader if isLoading and isChecked are set to true 1`] = `
.c1 {
  fill: #5DD167;
}

.c1.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c1.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c1 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c2 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c2.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c2 circle.outer {
  stroke: #5DD167;
}

.c2 circle.inner {
  stroke-width: 0;
  fill: #5DD167;
}

.c2.thin circle {
  stroke-width: 3;
}

.c2.absolute {
  position: absolute;
}

.c0 .visually-hidden {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  white-space: nowrap;
}

.c0.kirk-loader--inline {
  display: inline-block;
}

.c0.kirk-loader--fullScreen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba(255,255,255,0.85);
  height: 100vh;
  width: 100vw;
  z-index: 4;
}

.c0.kirk-loader--block {
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

.c0 .kirk-loader--done {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5DD167;
  color: #FFF;
  border-radius: 50%;
}

<div
  className="c0 kirk-loader--inline"
>
  <div
    aria-valuetext="Loading"
    role="progressbar"
  >
    <div
      className=""
      style={
        Object {
          "height": "24px",
          "width": "24px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="kirk-icon kirk-icon-circle spinning c1 c2"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="outer"
          cx="33"
          cy="33"
          fill="none"
          r="30"
        />
      </svg>
    </div>
  </div>
  <div
    className="visually-hidden"
    role="status"
  >
    Loading
  </div>
</div>
`;

exports[`RadioIcon Should render a loader if isLoading is set to true 1`] = `
.c1 {
  fill: #5DD167;
}

.c1.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c1.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c1 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c2 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c2.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c2 circle.outer {
  stroke: #5DD167;
}

.c2 circle.inner {
  stroke-width: 0;
  fill: #5DD167;
}

.c2.thin circle {
  stroke-width: 3;
}

.c2.absolute {
  position: absolute;
}

.c0 .visually-hidden {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  white-space: nowrap;
}

.c0.kirk-loader--inline {
  display: inline-block;
}

.c0.kirk-loader--fullScreen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba(255,255,255,0.85);
  height: 100vh;
  width: 100vw;
  z-index: 4;
}

.c0.kirk-loader--block {
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

.c0 .kirk-loader--done {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5DD167;
  color: #FFF;
  border-radius: 50%;
}

<div
  className="c0 kirk-loader--inline"
>
  <div
    aria-valuetext="Loading"
    role="progressbar"
  >
    <div
      className=""
      style={
        Object {
          "height": "24px",
          "width": "24px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="kirk-icon kirk-icon-circle spinning c1 c2"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="outer"
          cx="33"
          cy="33"
          fill="none"
          r="30"
        />
      </svg>
    </div>
  </div>
  <div
    className="visually-hidden"
    role="status"
  >
    Loading
  </div>
</div>
`;

exports[`RadioIcon Should render an empty circle by default 1`] = `
.c0 {
  fill: #00AFF5;
}

.c0.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c0.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c0 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c1.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c1 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c1.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c1 circle.outer {
  stroke: #00AFF5;
}

.c1 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c1.thin circle {
  stroke-width: 3;
}

.c1.absolute {
  position: absolute;
}

<svg
  aria-hidden={true}
  className="kirk-icon kirk-icon-circle c0 c1"
  height={24}
  viewBox="0 0 66 66"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  
  <circle
    className="outer"
    cx="33"
    cy="33"
    fill="none"
    r="30"
  />
</svg>
`;
